common/koa-middleware/withToggles.js

```diff
@@ -33,16 +39,44 @@ const parseCookies = function(req) {
 function withToggles(ctx, next) {
   const cookies = parseCookies(ctx.req);
   const togglesCookies = cookies.filter(cookie =>
-    cookie.key.startsWith('toggle_')
+    cookie.key.startsWith(cookiePrefix)
   );
   const toggles = togglesCookies.reduce((acc, cookie) => {
     return Object.assign({}, acc, {
-      [cookie.key.replace('toggle_', '')]: cookie.value === 'true',
+      [cookie.key.replace(cookiePrefix, '')]: cookie.value === 'true',
     });
   }, {});
 
   ctx.toggles = { ...defaultToggleValues, ...toggles };
+  enableDisableToggler(ctx);
   return next();
 }
 
+function enableDisableToggler(ctx) {
+  if (ctx && ctx.toggles) {
+    if (ctx.query.toggles) {
+      const reqFeatureToggle = ctx.query.toggles;
```

Comment:
The if statements here are throwing me a little bit.

What do you think about something like:

```ts
// This is toggle for now, we'll support multiple toggles later
const cookieName = `${cookiePrefix}${reqFeatureToggle}`;
const toggle = validToggle(ctx.query.toggles);
const state = toggle ? !toggle.startsWith('!') : false;

if (state === true) {
  ctx.cookies.set(cookieName, 'true', {
    maxAge: cookieExpiry,
  });
} else {
  ctx.cookies.set(cookieName, null); 
}
```

Reply:
even cleaner! yep